keras/backend/theano_backend.py

```diff
@@ -2840,3 +2841,77 @@ def local_conv2d(inputs, kernel, kernel_size, strides, output_shape, data_format
                          (output_row, output_col, -1, filters))
         output = permute_dimensions(output, (2, 0, 1, 3))
     return output
+
+
+def get_reachable_from_inputs(inputs, targets=None):
+    if targets is None:
+        raise NotImplementedError('It is not possible to use '
+                                  '`get_reachable_from_inputs` with the Theano'
+                                  ' backend without specifying target tensors to'
+                                  ' reach.')
+
+    backward_pass(targets)
+    return get_reachable_from_inputs_with_ptrs(inputs, targets)
+
+
+def backward_pass(outputs):
+    queue = outputs[:]
```

Comment:
What's the point of [:]?

Reply:
to copy the input list. We'll modify it later, and we don't want to modify the argument directly.